### PR TITLE
[Merged by Bors] - feat(LinearAlgebra/RootSystem/Basic): rm finiteness from def

### PR DESCRIPTION
**Pull request by ScottCarnahan**
There are many interesting infinite root systems, so we remove the finiteness assumption from the definition.  A few results use finiteness unnecessarily, so their proofs are revised.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/LinearAlgebra/RootSystem/Basic.lean [hunk 1]
```diff
@@ -148,6 +148,19 @@ lemma reflection_apply_self :
...
 
+@[simp]
+lemma reflection_same (x : M) : P.reflection i (P.reflection i x) = x :=
+  Module.involutive_reflection (P.coroot_root_two i) x
+
+lemma reflection_invOn_self : InvOn (P.reflection i) (P.reflection i) (range P.root)
+    (range P.root) := by
+  constructor
+  exact fun x _ => Module.involutive_reflection (P.coroot_root_two i) x
+  exact fun x _ => Module.involutive_reflection (P.coroot_root_two i) x
```
Comment (adomani) [inline Mathlib/LinearAlgebra/RootSystem/Basic.lean:159]:
```suggestion
  repeat exact fun x _ => Module.involutive_reflection (P.coroot_root_two i) x
```
if you like it!

File: Mathlib/LinearAlgebra/RootSystem/Basic.lean [hunk 2]
```diff
@@ -148,6 +148,19 @@ lemma reflection_apply_self :
     P.reflection i (P.root i) = - P.root i :=
   Module.reflection_apply_self (P.coroot_root_two i)
 
+@[simp]
+lemma reflection_same (x : M) : P.reflection i (P.reflection i x) = x :=
```
Comment (adomani) [inline Mathlib/LinearAlgebra/RootSystem/Basic.lean:152]:
```suggestion
lemma reflection_self (x : M) : P.reflection i (P.reflection i x) = x :=
```
maybe?  I am not a naming expert.

**Review by adomani — APPROVED** [review]
Looks very nice, thanks!

File: Mathlib/LinearAlgebra/RootSystem/Basic.lean (hunk 1 above)
Comment (jcommelin) [inline Mathlib/LinearAlgebra/RootSystem/Basic.lean:159]: Alternatively
```suggestion
  constructor <;>
    exact fun x _ => Module.involutive_reflection (P.coroot_root_two i) x
```

File: Mathlib/LinearAlgebra/RootSystem/Basic.lean [hunk 3]
```diff
@@ -156,15 +169,23 @@ lemma coreflection_apply (f : N) :
     P.coreflection i f = f - (P.toLin (P.root i) f) • P.coroot i :=
   rfl
 
-lemma bijOn_reflection_root :
-    BijOn (P.reflection i) (range P.root) (range P.root) :=
-  ((finite_range P.root).injOn_iff_bijOn_of_mapsTo (P.mapsTo_preReflection_root i)).mp <|
-    (P.reflection i).injective.injOn _
+  @[simp]
```
Comment (jcommelin) [inline Mathlib/LinearAlgebra/RootSystem/Basic.lean:172]:
```suggestion
@[simp]
```

File: Mathlib/LinearAlgebra/RootSystem/Basic.lean [hunk 4]
```diff
@@ -156,15 +169,23 @@ lemma coreflection_apply (f : N) :
...
+lemma coreflection_apply_self :
+    P.coreflection i (P.coroot i) = - P.coroot i :=
+  Module.reflection_apply_self (P.flip.coroot_root_two i)
+
+lemma coreflection_eq_flip_reflection (f : N) : P.coreflection i f = P.flip.reflection i f :=
+  rfl
+
+@[simp]
+lemma coreflection_same (x : N) : P.coreflection i (P.coreflection i x) = x :=
```
Comment (jcommelin) [inline Mathlib/LinearAlgebra/RootSystem/Basic.lean:181]: We generally use `self` when a variable `x` occurs in two places, and `of_same` when there is `i`, `j` and an assumption `h : i = j`.
```suggestion
lemma coreflection_self (x : N) : P.coreflection i (P.coreflection i x) = x :=
```

**Review by jcommelin — COMMENTED** [review]
bors d+

**Pull request closed without merging**